openedx/features/discussion_provider_piazza/views.py

```diff
@@ -0,0 +1,170 @@
+        """
+        Returns the LTI embed code for embedding in the current course context.
+        Args:
+            course_key (CourseKey): Course key for course in which to embed LTI.
+            request (HttpRequest): Request object for view in which LTI will be embedded.
+
+        Returns:
+            HTML code to embed LTI in course page.
+
+        """
+        discussion_config = get_course_discussion_config(course_key)
+        lti_creds = self.get_oauth_credentials(discussion_config)
+        user_id = self.get_user_id(request.user, course_key)
+        context_id = self.get_context_id(course_key)
+        resource_link_id = self.get_resource_link_id(course_key, request)
+        roles = self.get_lti_roles(request.user, course_key)
+        context_title = self.get_context_title(course_key)
+        result_sourcedid = self.get_result_sourcedid(context_id, resource_link_id, user_id)
+        additional_params = self.get_additional_lti_parameters(course_key, request)
+
+        return lti_embed(
+            html_element_id='discussion-provider-lti-launcher',
+            lti_launch_url=self.LTI_LAUNCH_URL,
+            oauth_key=lti_creds.oauth_key,
+            oauth_secret=lti_creds.oauth_secret,
+            resource_link_id=resource_link_id,
+            user_id=user_id,
+            roles=roles,
+            context_id=context_id,
+            context_title=context_title,
+            context_label=context_id,
+            result_sourcedid=result_sourcedid,
+            **additional_params,
+        )
+
+    def render_to_fragment(self, request: HttpRequest, course_id: str) -> Fragment:
+        """
+        Returns a fragment view for the LTI launch.
+        Args:
+            request (HttpRequest): request object
+            course_id (str): A string course id
+
+        Returns:
+            A Fragment that embeds LTI in a course page.
+
+        """
+        course_key = CourseKey.from_string(course_id)
+        lti_embed_html = self.get_lti_embed_code(course_key, request)
+
+        fragment = Fragment(
+            HTML(
+                """
+                <iframe
+                    id='discussion-provider-lti-embed'
+                    srcdoc='{}'
+                 >
+                </iframe>
+                """
+            ).format(lti_embed_html)
+        )
+        fragment.add_css(
+            """
+            #discussion-provider-lti-embed {
+                width: 100%;
+                min-height: 400px;
+                border: none;
+            }
+            """
+        )
+        return fragment
+
+
+class PiazzaCourseTabView(LtiCourseLaunchMixin, EdxFragmentView):
+    """
+    Course tab view for Piazza discusion provider.
+    """
+    LTI_LAUNCH_URL = "https://piazza.com/connect"
+
+    @staticmethod
+    def get_oauth_credentials(config):
+        return LtiCredentials(
+            config.config["consumer_key"],
+            config.config["consumer_secret"],
+        )
```

Comment:
Eventually this will be all there is in the  plugin, the rest will move to openedx.core ...